core/block/editor/widget/widget.go

```diff
@@ -106,3 +107,12 @@ func (w *widget) CreateBlock(s *state.State, req *pb.RpcBlockCreateWidgetRequest
 
 	return wrapper.Model().Id, nil
 }
+
+func (w *widget) CreateBlocks(s *state.State, reqs []*pb.RpcBlockCreateWidgetRequest) error {
```

Comment:
don't over-complicate api

Reply:
@deff7 I was solving the problem with single Apply on request. Previous version used to invoke CreateWidgetBlock from 4 to 6 times, that triggers state.Apply every time and rewites the list of events. The main aim here is to send events on creating Widget blocks synchronously, so either we should check messages list in context inside Apply, or create separate command that will create multiple blocks and invoke Apply one single time

Reply:
We need some kind of bulking ops tho

Reply:
CreateBlock doesn't apply changes, DoStateCtx does